hail/python/hail/experimental/plots.py

```diff
@@ -264,3 +269,148 @@ def scale_file_sizes(file_sizes):
+
+        #Partition each genotype call
+        GT_HET = mt_variant_df.loc[mt_variant_df['GTA'] == '1/0']
+        GT_HOM_DOM = mt_variant_df.loc[mt_variant_df['GTA'] == '0/0']
+        GT_HOM_R = mt_variant_df.loc[mt_variant_df['GTA'] == '1/1']
+        GT_NA = mt_variant_df.loc[mt_variant_df['GTA'] == './.']
+
+        #For ellipse : generate mean angle for each GT partition
+        HET_THETA = GT_HET.THETA.mean()
+        HOM_DOM_THETA = GT_HOM_DOM.THETA.mean()
+        HOM_R_THETA = GT_HOM_R.THETA.mean()
+
+        def max_distance(values):
+            v = max(values)-min(values)
+            return v
+
+        #For ellipse : Take [1% to 99% Quantile] of values, this minimizes noise
+        HET_QUANT_X = GT_HET.NORMX.quantile([.01, .99])
+        HET_QUANT_Y = GT_HET.NORMY.quantile([.01, .99])
+        HOM_DOM_QUANT_X = GT_HOM_DOM.NORMX.quantile([.01, .99])
+        HOM_DOM_QUANT_Y = GT_HOM_DOM.NORMY.quantile([.01, .99])
+        HOM_R_QUANT_X = GT_HOM_R.NORMX.quantile([.01, .99])
+        HOM_R_QUANT_Y = GT_HOM_R.NORMY.quantile([.01, .99])
+
+        #For ellipse : take maximum distance between the genotype calls once noise is removed
+        #this gives width and height for ellipse
+        HET_MAX_X = max_distance(HET_QUANT_X)
+        HET_MAX_Y = max_distance(HET_QUANT_Y)
+        HOM_DOM_MAX_X = max_distance(HOM_DOM_QUANT_X)
+        HOM_DOM_MAX_Y = max_distance(HOM_DOM_QUANT_Y)
+        HOM_R_MAX_X = max_distance(HOM_R_QUANT_X)
+        HOM_R_MAX_Y = max_distance(HOM_R_QUANT_Y)
+
+        df = pd.DataFrame(
+            {
+                "GT_call": mt_variant_df['GTA'],
+                "HET_XNORMS": GT_HET['NORMX'],
+                "HET_YNORMS": GT_HET['NORMY'],
+                "HOM_DOM_XNORMS": GT_HOM_DOM['NORMX'],
+                "HOM_DOM_YNORMS": GT_HOM_DOM['NORMY'],
+                "HOM_R_XNORMS": GT_HOM_R['NORMX'],
+                "HOM_R_YNORMS": GT_HOM_R['NORMY'],
+                "GT_NA_YNORMS": GT_NA['NORMX'],
+                "GT_NA_XNORMS": GT_NA['NORMY'],
+                "BAF" : mt_variant_df['BAF'],
+                "LRR" : mt_variant_df['LRR'],
+                "HET_X" : HET_MEAN_X,
+                "HET_Y": HET_MEAN_Y,
+                "HET_THETA" : HET_THETA,
+                "HET_W" : HET_MAX_X,
+                "HET_H" : HET_MAX_Y,
+                "HOM_DOM_X" : HOM_DOM_MEAN_X,
+                "HOM_DOM_Y": HOM_DOM_MEAN_Y,
+                "HOM_DOM_THETA" : HOM_DOM_THETA,
+                "HOM_DOM_W" : HOM_DOM_MAX_X,
+                "HOM_DOM_H" : HOM_DOM_MAX_Y,
+                "HOM_R_X" : HOM_R_MEAN_X,
+                "HOM_R_Y": HOM_R_MEAN_Y,
+                "HOM_R_THETA" : HOM_R_THETA,
+                "HOM_R_W" : HOM_R_MAX_X,
+                "HOM_R_H" : HOM_R_MAX_Y
+            }
+        )
+        df
+
+        source = ColumnDataSource.from_df(df)
+        hover = HoverTool(
+            tooltips=[
+                ('Sample ID', '@Sample_ID'),
+                ('GT Call', '@GT_call')
```

Comment:
I have struggled with how to do this -- since the sample id and GT call I didn't partition by GT so I call it directly from mt_variant_df. If I were to do that, I would need to add the "parent" normX and NormY values in addition to their GT specific lists within the dictionary (meaning define mt_variant_df['NORMX'] and mt_variant_df['NORMY'].